x/evm/keeper/statedb.go

```diff
@@ -79,27 +72,28 @@ func (k *Keeper) ForEachStorage(
 func (k *Keeper) SetAccBalance(
 	ctx sdk.Context, addr gethcommon.Address, amountEvmDenom *big.Int,
 ) error {
-	nativeAddr := sdk.AccAddress(addr.Bytes())
-	balance := k.bankKeeper.GetBalance(ctx, nativeAddr, evm.EVMBankDenom).Amount.BigInt()
+	addrBech32 := eth.EthAddrToNibiruAddr(addr)
+	balance := k.Bank.GetBalance(ctx, addrBech32, evm.EVMBankDenom).Amount.BigInt()
 	delta := new(big.Int).Sub(amountEvmDenom, balance)
+	bk := k.Bank.BaseKeeper
```

Comment:
The `Bank.BaseKeeper` needs to be used here because we're inside the setter that the `StateDB` uses to commit balances. Using the `NibiruBankKeeper`'s overrides would double-label and mutate the journal while its being committed, leading to strange behavior.